mlflow/tracing/fluent.py

```diff
@@ -549,3 +550,133 @@ def f():
+        )
+
+    if trace.info.status not in TraceStatus.end_statuses():
+        raise MlflowException.invalid_parameter_value("The remote trace must be ended.")
+
+    if active_span := get_current_active_span():
+        # If there is an active span, merge the new trace under the active one
+        _merge_remote_trace_to_local(
+            remote_trace=trace,
+            local_request_id=active_span.request_id,
+            local_parent_span_id=active_span.span_id,
+        )
+    else:
+        # If there is no active span, create a new root span named "Remote Trace <...>"
+        # and put the remote trace under it. This design aims to keep the trace export
+        # logic simpler and consistent, rather than directly exporting the remote trace.
+        client = MlflowClient()
+        remote_root_span = trace.data.spans[0]
+        span = client.start_trace(
+            name=f"Remote Trace <{remote_root_span.name}>",
+            inputs=remote_root_span.inputs,
+            attributes={
+                # Exclude request ID attribute not to reuse same request ID
+                k: v
+                for k, v in remote_root_span.attributes.items()
+                if k != SpanAttributeKey.REQUEST_ID
+            },
+            start_time_ns=remote_root_span.start_time_ns,
+        )
+        _merge_remote_trace_to_local(
+            remote_trace=trace,
+            local_request_id=span.request_id,
+            local_parent_span_id=span.span_id,
+        )
+        client.end_trace(
+            request_id=span.request_id,
+            status=trace.info.status,
+            outputs=remote_root_span.outputs,
+            end_time_ns=remote_root_span.end_time_ns,
+        )
+
+
+def _merge_remote_trace_to_local(
+    remote_trace: Trace,
+    local_request_id: str,
+    local_parent_span_id: str,
+):
+    """
+    Merge the given trace object under an existing trace in the in-memory trace registry.
+
+    This is particularly used for handling a remote trace that is created in a different
+    process and needs to be merged into the current trace.
+
+    Args:
+        trace: The trace object to be merged.
+        parent_request_id: The request ID of the parent trace.
+        parent_span_id: The parent span ID, under which the child trace should be merged.
+    """
+    trace_manager = InMemoryTraceManager.get_instance()
+
+    # The merged trace should have the same trace ID as the parent trace.
+    with trace_manager.get_trace(local_request_id) as parent_trace:
+        if not parent_trace:
+            _logger.warning(
+                f"Parent trace with request ID {local_request_id} not found. Skipping merge."
+            )
+        new_trace_id = parent_trace.span_dict[local_parent_span_id]._trace_id
+
+    # NB: We clone span one by one in the order it was saved in the original trace. This
+    # works upon the assumption that the parent span always comes before its children.
```

Comment:
Can we check if an async trace system (Arize?) will export a trace in insertion order for async collection or if it is sorted prior to export?
If unordered, we will likely need to warn users in the docs about this (to use the API, they should validate the trace data prior to constructing a `Trace` object from a response).

Reply:
What you think will be the use case for integrating with other async trace system? The reason I'm asking is, this API only support merging a trace that strictly conform to our spec such as trace info, reserved attributes e.g. `mlflow.traceRequestID`, and MLflow trace should have sorted spans by design.

Reply:
I'm playing Devil's advocate here and imagining a user calling a service that has an internal async processing framework that is returning a synchronized response. If we're not sure what the behavior of that external service is with adhering to ordering, we might have an interesting data structure to deal with. 
I'm not saying we should support this - rather that we mention in the API docs (just in the docstring - no need to put low-level details like this in the actual docs for tracing) for this API that we are assuming that the trace is ordered and offer no guarantees of proper operation if the spans within the external trace are not sorted properly.

Reply:
I see, yea the order might be broken for some edge cases. Added a docstring and also enrich the exception message when we got non-sorted trace.